Client/TabManagement/TabManagerImplementation.swift

```diff
@@ -110,7 +129,6 @@ class TabManagerImplementation: LegacyTabManager {
     override func preserveTabs() {
         // For now we want to continue writing to both data stores so that we can revert to the old system if needed
         super.preserveTabs()
-        guard shouldUseNewTabStore() else { return }
```

Comment:
This guard isn't needed anymore?

Reply:
Nop because we want to preserve the data on both "systems" in case we need to rollback to the existing tab storage

Reply:
Actually can we put this back for now? The issue with the homepage might mean the migration needs to be done again which means peoples dev environments might be broken. Not the end of the world if we have to tell people to reinstall but better if we don't have to. We can remove it in the same PR that turns on the flag.